cmd2/cmd2.py

```diff
@@ -185,6 +163,8 @@ def with_argument_list(func: Callable) -> Callable:
     method. Default passes a string of whatever the user typed.
     With this decorator, the decorated method will receive a list
     of arguments parsed from user input using shlex.split()."""
+    import functools
```

Comment:
I don't know enough about how Python importing works ...

If we call this function 1000 times, it just gets imported once right?  i.e. there isn't a performance penalty here is there?

Also, if we then call the function below and hit the same import on line 184 below after calling this function, it doesn't import it again there right?

Reply:
Correct. https://docs.python.org/3/reference/import.html#the-module-cache describes how the import machinery first looks to see if the module is already imported, if so, it just uses the module already in the cache. Think of "import functools" as "import functools if it hasn't already been imported"